x-pack/plugins/security_solution/public/detections/components/rules/rule_status/index.tsx

```diff
@@ -55,6 +59,19 @@ const RuleStatusComponent: React.FC<RuleStatusProps> = ({ ruleId, ruleEnabled })
     }
   }, [fetchRuleStatus, ruleId]);
 
+  const getStatus = useCallback((status: RuleStatusType | null | undefined) => {
+    if (status == null) {
+      return getEmptyTagValue();
+    } else if (status != null && status === 'partial failure') {
+      // Temporary fix if on upgrade a rule has a status of 'partial failure' we want to display that text as 'warning'
+      // On the next subsequent rule run, that 'partial failure' status will be re-written as a 'warning' status
+      // and this code will no longer be necessary
+      // TODO: remove this code in 8.0.0
+      return 'warning';
+    }
```

Comment:
Disclaimer: I'm not familiar with Kibana upgrades and migrations, what's supported and not.

What if a user upgrades Kibana from let's say 7.10 (where a rule can have a `partial failure` status) to 8.1? I mean data-wise. Such a rule will (for a short time) have a status which will not be supported in the API anymore (do I understand the intention right?). But still we'll fetch this `partial failure` status, return in the API response, and the client will receive it. And it's ok if this "adapter"-like code will still be here, but if not, we might get something we don't want in the UI.

Wouldn't it be simpler to do either of the following:

- render `partial failure` as `"warning"` in the UI layer without modifying the contract between the client and server
- add a new `warning` status and convert all partial failures to it on the server; and support it on the client

I'm also not sure about the current users of our API endpoints. While this is not a breaking change in terms of response structure, it's a change in behaviour. If some code in userland relies on some checks for `partial failure`, that could be a de-facto breaking change.

Reply:
Considering user-land breaking changes we will have this change reflected in the docs and in the release notes. If a status is 'partial failure' after upgrade it will still be reflected as a 'warning' on the UI so we do take care of option 1.

Option 2 we no longer write 'partial failure' as a status so these statuses will technically be migrated off of 'partial failure' and to 'warning' as the status text on the server side. You bring up a good point though that even with 8.0 allowing breaking changes this code should still be left in there to account for situations like that. 

My hope is that by 8.0 we will be completely off of the saved objects used for rule statuses and then this code won't really be relevant anymore.